File: dotnet/net6/Watches-v6/Watch.cs
namespace Watches;

public class Watch
{
    private readonly DateTime _time;
    private readonly IDisplayStrategy _display;
    private readonly IAlarmStrategy _alarm;

    public Watch(IDisplayStrategy displayStrategy, IAlarmStrategy alarmStrategy) : this(DateTime.Now, displayStrategy, alarmStrategy) {}
    
    public Watch(DateTime time, IDisplayStrategy displayStrategy, IAlarmStrategy alarmStrategy)
    {
        _time = time;
        _display = displayStrategy;
        _alarm = alarmStrategy;
    }

    public void DisplayTime()
    {
        _display.Display(_time);
    }

    public void DisplayAlarms()
    {
        foreach (var alarm in _alarm.GetAlarms())
        {
            Console.WriteLine(alarm);
        }
    }

    public void AddAlarm(Alarm alarm)
    {
        _alarm.AddAlarm(alarm);
    }
}

File: python/watch.py
class Watch():
    def __init__(self, time, display_strategy, alarm_strategy):
        self._time = time
        self._display_strategy = display_strategy
        self._alarm_strategy = alarm_strategy

    def display_time(self):
        self._display_strategy.display(self._time)

    def display_alarms(self):
        for alarm in self._alarm_strategy.get_alarms():
            print(alarm)

    def add_alarm(self, alarm):
        self._alarm_strategy.add_alarm(alarm)
